src/cargo/util/toml/mod.rs

```diff
@@ -1342,7 +1342,7 @@ impl TomlManifest {
             .unwrap()
             .clone();
         package.workspace = None;
-        package.resolver = ws.resolve_behavior().to_manifest();
+        package.resolver = Some(ws.resolve_behavior().to_manifest());
```

Comment:
One thought I had was to do something like

```rust
        if ws.resolve_behavior() != package.resolver.unwrap_or(resolver_from(package.edition)) {
            package.resolver = Some(ws.resolve_behavior().to_manifest());
        }
```

so that it's only explicitly set when needed